RFCs/0000-add-pdk-config.md

```diff
@@ -0,0 +1,292 @@
+
+    - Known single value key, update value:
+
+      ```
+      $ pdk config set user.default_template.url "https://github.com/example/repo.git"
+      pdk (INFO): Changed existing value of user.default_template.url from "..." to "..."
+      https://github.com/example/repo.git
+      ```
+
+    - Known multi-value key, initial value:
+
+      ```
+      $ pdk config set module.multi.example apple
+      pdk (INFO): Added new value "apple" to module.multi.example
+      apple
+      ```
+
+    - Known multi-value key, additional value:
+
+      ```
+      $ pdk config set module.multi.example banana
+      pdk (INFO): Added new value "banana" to module.multi.example
+      apple
+      banana
+      ```
+
+    - Known multi-value key, duplicate value:
+
+      ```
+      $ pdk config set module.multi.example apple
+      pdk (INFO): No changes made to module.multi.example as it already contains value "apple"
+      apple
+      banana
+      ```
+
+    - User-defined key, initial value:
+
+      ```
+      $ pdk config set module.x.example strawberry
+      pdk (INFO): Set initial value of modle.x.example to "strawberry"
+      strawberry
+      ```
+
+    - User-defined key, append value:
+
+      ```
+      $ pdk config set --add module.x.example orange
+      pdk (INFO): Added new value "orange" to module.x.example
+      strawberry
+      orange
+      ```
+
+  - `pdk config del[ete] <key> [<value>|--all]`
+
+    Unset one more more values from the given configuration key.
+
+    If `<key>` is not a leaf node, exits non-zero with an error message.
+
+    If `<key>` currently stores no value, issues a warning but exits with a success status.
+
+    If `<key>` currently stores a single value:
+
+      - If no `<value>` is provided, unsets existing value.
+
+      - If provided `<value>` matches existing value, unsets existing value.
+
+      - If provided `<value>` does not match existing value, exits non-zero with an error message.
+
+      - If passed `--all` in place of a value, unsets existing value.
+
+    If `<key>` currently stores multiple values:
+
+      - If no `<value>` is provided, exit non-zero with an error message indicating user must pass `--all` if they want
+        to clear all values for the key.
+
+      - If provided `<value>` is present in the list of values for `<key>`, removes value from list.
+
+      - If provided `<value>` is not present in the list of values for `<key>`, issues a warning but exits with a
+        success status.
+
+      - If passed `--all` in place of a value, empties the list of all existing values.
+
+### Proposed Configuration Keys
+
+#### User/Local Config
+
+The values for these config keys will be persisted in ~/.pdk/config. They are intended to be developer-specific settings
+and will not apply to another developer working on the same codebase.
+
+| Key | Description |
+| --- | --- |
+| user.module\_defaults.template.url | Default template URL to use when generating a new module. |
```

Comment:
`module_defaults.template.*`  values will also be used for `pdk convert`. To change these defaults, you would use `pdk config` again. Specifying `--template-url` will not change the defaults, as they currently do in the `answers.json`.